### DISCUSSION TITLE
How to set the styles of parameter tree children?

### DISCUSSION THREAD

**Discussion by philipqueen** [pyqtgraph/pyqtgraph#2677] [Q&A] [answered]
Hi! 

I'm working on a GUI that uses the `pyqtgraph.parametertree` to input parameters. The problem is that without specifying the styles for the children of the tree in the css, the darkmode preference on my machine is making the input boxes illegible (pictures below). The other problems I've had with darkmode inheritance have been solved by just explicitly setting the colors for the elements I've had trouble with, but I can't figure out how to set the css for these children elements.

Is there a way to specifically refer to these elements in the css? Or another way to style them?

Thanks for any help!

A "list" type child element, text is only visible when hovering:
<img width="191" alt="Screen Shot 2023-04-05 at 11 57 27 AM" src="https://user-images.githubusercontent.com/24758117/230166884-dab6c54e-2406-43df-9418-5e1214361e67.png">
A "float" type child element, text is only visible when highlighting:
<img width="150" alt="Screen Shot 2023-04-05 at 11 57 47 AM" src="https://user-images.githubusercontent.com/24758117/230166921-bd3083c7-8360-4fa0-a530-0ed8760ce5d5.png">

**Comment by ntjess** [comment]
Unfortunately Qt doesn't respect dark mode on windows so I can't test this, but can you try messing with `GroupParameterItem.updateDepth` and seeing if that impacts the coloring?

**Reply by j9ac9k** [reply]
Qt might respect dark mode on Windows with Qt 6.5 ... I think I saw something about that in the changelog

**Reply by philipqueen** [reply]
Thanks for the response @ntjess! 

Unfortunately I haven't been able to get `GroupParameterItem.updateDepth` working. The closest I've been able to come is doing `self._parameter_group.itemClass.updateDepth(depth=0)` on a GroupParameter object, which gives the error `TypeError: updateDepth() missing 1 required positional argument: 'self'`. I used `itemClass` based on [this in the documentation](https://pyqtgraph.readthedocs.io/en/latest/api_reference/parametertree/parametertypes.html#pyqtgraph.parametertree.parameterTypes.GroupParameter.itemClass). Every other attempt I've made has lead to an attribute error like: `AttributeError: 'ParameterTree' object has no attribute 'parameterTypes'`.

I apologize if I'm making some simple mistakes, I'm not very familiar with these parameter trees.

**Reply by ntjess** [reply]
> Qt might respect dark mode on Windows with Qt 6.5 ... I think I saw something about that in the changelog

Found the blog post: https://www.qt.io/blog/dark-mode-on-windows-11-with-qt-6.5

Neat! With `-style fusion`:
<img width="564" alt="image" src="https://user-images.githubusercontent.com/23620506/230982619-765b515d-7006-4cab-bf99-e53a8525bf0f.png">

It doesn't seem to replicate the dark mode issues for me, though

@philipqueen Apologies for my cryptic answer, I was suggesting something like this:
```python
from pyqtgraph.parametertree.parameterTypes.basetypes import GroupParameterItem
# This should happen before you spawn any tree widgets
GroupParameterItem.updateDepth = lambda *args: None

# Rest of your widget code here
```

Is your issue still appearing with this change? I would try to investigate further but it is difficult for me to replicate on my machine

**Reply by philipqueen** [reply]
Thanks for the explicit code! I tried a few different configurations of it without any luck on changing the colors though. From the source code it looks like it should be able to change the background color.

I appreciate the help,  I understand it's not easy fixing something like this from afar

**Comment by philipqueen** [comment, marked as answer]
A few months later, and I finally figured out a solution to this. The relevant PyQtGraph object is `Parameter`, not `ParameterItem`. This means the most specific type of this you can reference is `QObject`. This means you can override the systems darkmode setting by doing something like `QObject { color: black, background-color: white}` in your CSS. Be careful, this should be very early in your css, as it will override any other QObject derived objects that came before it. So make sure your styling of `QMainWindow` or `QGroupBox`, for example, come after the `QObject` styling.

Thanks again to everyone who helped with this
